### PR TITLE
Added a New Index Soil Water Index  SWI

### PR DESCRIPTION
**Pull request by MATRIX4284**
Added SWI 

Reference :
https://land.copernicus.eu/global/products/swi

### CODE REVIEW

**Comment by adamjstewart** [comment]
It looks like you replaced the existing `AppendNDWI` instead of adding a new `AppendSWI`. Maybe a bad rebase?

**Comment by MATRIX4284** [comment]
> It looks like you replaced the existing `AppendNDWI` instead of adding a new `AppendSWI`. Maybe a bad rebase?sorry 

Sorry deleted it by mistake ..Fixed it

**Comment by MATRIX4284** [comment]
Fixed the error in which i deleted an existing function

**Comment by kaushikCanada** [comment]
this is a great feature!

File: torchgeo/transforms/indices.py [hunk 1]
```diff
@@ -147,21 +147,34 @@ def __init__(self, index_red: int, index_nir: int) -> None:
...
             index_nir: index of the Near Infrared (NIR) band in the image
         """
         super().__init__(index_a=index_green, index_b=index_nir)
+
+class AppendSWI(AppendNormalizedDifferenceIndex):
+    """Normalized Difference Water Index (NDWI).
+
+    If you use this index in your research, please cite the following paper:
+
+    * https://ui.adsabs.harvard.edu/abs/2020ISPAn.5.3...33J/abstract#:~:text=Sentinel%2D2%20is%20a%20new,higher%20spatial%20and%20spectral%20resolution.&text=The%20results%20showed%20that%20overall,water%20for%20Sentinel%2D2%20data.
```
Comment (calebrob6) [inline torchgeo/transforms/indices.py:170]: Is this the correct citation? The format of the title is very strange. I also found this paper that might be more applicable -- https://www.mdpi.com/2073-4441/13/12/1647/htm?

File: torchgeo/transforms/indices.py [hunk 2]
```diff
@@ -147,21 +147,34 @@ def __init__(self, index_red: int, index_nir: int) -> None:
...
         """Initialize a new transform instance.
-
         Args:
             index_green: index of the Green band in the image
             index_nir: index of the Near Infrared (NIR) band in the image
         """
         super().__init__(index_a=index_green, index_b=index_nir)
+
+class AppendSWI(AppendNormalizedDifferenceIndex):
+    """Normalized Difference Water Index (NDWI).
```
Comment (calebrob6) [inline torchgeo/transforms/indices.py:166]: Needs to be changed to SWI

File: torchgeo/transforms/indices.py [hunk 3]
```diff
@@ -147,21 +147,34 @@ def __init__(self, index_red: int, index_nir: int) -> None:
...
     """Normalized Difference Water Index (NDWI).
-
     If you use this index in your research, please cite the following paper:
-
     * https://doi.org/10.1080/01431169608948714
     """
 
     def __init__(self, index_green: int, index_nir: int) -> None:
         """Initialize a new transform instance.
-
```
Comment (calebrob6) [inline torchgeo/transforms/indices.py:162]: Can you reset all these deleted spaces to how they previously were?

File: torchgeo/transforms/indices.py [hunk 4]
```diff
@@ -147,21 +147,34 @@ def __init__(self, index_red: int, index_nir: int) -> None:
...
+
+    * https://ui.adsabs.harvard.edu/abs/2020ISPAn.5.3...33J/abstract#:~:text=Sentinel%2D2%20is%20a%20new,higher%20spatial%20and%20spectral%20resolution.&text=The%20results%20showed%20that%20overall,water%20for%20Sentinel%2D2%20data.
+    """
+    
+    def __init__(self, index_red: int, index_swir: int) -> None:
+        """Initialize a new transform instance.
+
+        Args:
+            index_green: index of the Green band in the image
+            index_nir: index of the Near Infrared (NIR) band in the image
```
Comment (calebrob6) [inline torchgeo/transforms/indices.py:178]: should be `index_swir` with appropriate text

**Review by calebrob6 — CHANGES_REQUESTED** [review]

File: torchgeo/transforms/indices.py (hunk 1 above)
Comment (MATRIX4284) [reply torchgeo/transforms/indices.py]: citation corrected.
https://www.droughtmanagement.info/literature/ISPRS_Various_Drought_Indices_Monitoring_Drought_Condition_Aravalli_Terrain_India_2004.pdf ..
This is the actual paper written by C. Bhuiyan
Department of Civil Engineering, Indian Institute of Technology Kanpur
Kanpur-208016, India, bhuiyan@iitk.ac.in

**Comment by calebrob6** [comment]
Hey @MATRIX4284, thanks for the contribution! I just looked further into this and it seems that the band index described in the paper you cited (https://doi.org/10.5194/isprs-annals-V-3-2020-33-2020
) is different than the "SWI" calculated by Copernicus Global Land Service (https://land.copernicus.eu/global/products/SWI).

I'm OK with adding this index, but I think we should make it clear in the documentation / method docstring that these are different things.

File: torchgeo/transforms/indices.py (hunk 2 above)
Comment (MATRIX4284) [reply torchgeo/transforms/indices.py]: Changed the name and comments accordingly

File: torchgeo/transforms/indices.py (hunk 4 above)
Comment (MATRIX4284) [reply torchgeo/transforms/indices.py]: changed

**Comment by MATRIX4284** [comment]
> Hey @MATRIX4284, thanks for the contribution! I just looked further into this and it seems that the band index described in the paper you cited (https://doi.org/10.5194/isprs-annals-V-3-2020-33-2020 ) is different than the "SWI" calculated by Copernicus Global Land Service (https://land.copernicus.eu/global/products/SWI).
> 
> I'm OK with adding this index, but I think we should make it clear in the documentation / method docstring that these are different things.

sure ..i have already added in the comments ..i will update the documentation .

File: torchgeo/transforms/indices.py (hunk 1 above)
Comment (MATRIX4284) [reply torchgeo/transforms/indices.py]: True ..i reviewed and found that  https://www.mdpi.com/2073-4441/13/12/1647/html is more applicable..
i will post a PR changing it asap ..thanks for the guidance.

Comment (MATRIX4284) [reply torchgeo/transforms/indices.py]: Pushed it with the correct citation https://www.mdpi.com/2073-4441/13/12/1647/html

File: Dockerfile_ubuntu_1804_cuda_11_3_cudnn_8_torchgeo [hunk 5]
```diff
@@ -0,0 +1,164 @@
+FROM nvidia/cuda:11.3.1-cudnn8-devel-ubuntu18.04
```
Comment (adamjstewart) [inline Dockerfile_ubuntu_1804_cuda_11_3_cudnn_8_torchgeo:1]: I'm guessing this was added by accident?

Comment (MATRIX4284) [reply Dockerfile_ubuntu_1804_cuda_11_3_cudnn_8_torchgeo:1]: yes i need to roll this back ..and add it as a separate PR head..i wanted to create a different header for this PR but it got pushed to SWI PR

Comment (adamjstewart) [reply Dockerfile_ubuntu_1804_cuda_11_3_cudnn_8_torchgeo]: I recommend creating a separate branch for each PR, that way changes don't start to end up in multiple PRs

Comment (MATRIX4284) [reply Dockerfile_ubuntu_1804_cuda_11_3_cudnn_8_torchgeo]: thanks

**Comment by calebrob6** [comment]
@MATRIX4284 can you make all of the new index related contributions in a single pull request? The other pull request has a duplicate of the changes that are made here.

**Comment by MATRIX4284** [comment]
Close PR

**Pull request closed without merging**